### PR TITLE
Test: Add Unit tests for params under event parsing

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
## Initial Checklist

- [x] There is an issue describing the need for this PR.
- [x] Git log contains summary of the change.
- [x] Git log contains motivation and context of the change.
- [x] If part of an EPIC, PR git log contains EPIC number.
- [ ] If part of an EPIC, PR was added to EPIC description.

## Description (git log)

Added unit tests for the argument parsing in tracee-rules:
[pkg/events/parse/params.go](https://github.com/aquasecurity/tracee/blob/main/pkg/events/parse/params.go)

Fixes: #2196 

## Type of change

- [ ] Bug fix (non-breaking change fixing an issue, preferable).
- [ ] Quick fix (minor non-breaking change requiring no issue, use with care)
- [ ] Code refactor (code improvement and/or code removal)
- [x] New feature (non-breaking change adding functionality).
- [ ] Breaking change (cause existing functionality not to work as expected).

## How Has This Been Tested?
```sh
cd pkg/events/parse
go test -v
```
Tests being included in this PR:

- [x] [params_test.go](https://github.com/ShubhamPalriwala/tracee/blob/eb166efc5cc8f4603a8feb8ccad0e618206da8ab/pkg/events/parse/params_test.go)

Reproduce the test by running:

- cd pkg/events/parse
- go test -v

## Final Checklist:

Pick "Bug Fix" or "Feature", delete the other and mark appropriate checks.

- [ ] I have made corresponding changes to the documentation.
- [x] My code follows the style guidelines (C and Go) of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented all functions/methods created explaining what they do.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix, or feature, is effective.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published before.

## Git Log Checklist:

My commits logs have:

- [ ] Subject starts with "subsystem|file: description".
- [x] Do not end the subject line with a period.
- [x] Limit the subject line to 50 characters.
- [x] Separate subject from body with a blank line.
- [x] Use the imperative mood in the subject line.
- [x] Wrap the body at 72 characters.
- [x] Use the body to explain what and why instead of how.

### CODE REVIEW

File: pkg/events/parse/params_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
+package parse
+
+import (
+	"testing"
+
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+func TestArgInt32Val(t *testing.T) {
+	e := trace.Event{Args: []trace.Argument{
```
Comment (josedonizetti) [inline pkg/events/parse/params_test.go:10]: @ShubhamPalriwala can we use the table test here and check for the 3 cases of each method?

- valid value (we should check the return value)
- invalid argument type
- argument not found

**Review by josedonizetti — COMMENTED** [review]
@ShubhamPalriwala Thank you for looking into it. I know you are starting with golang, so let me know if you have any questions. An example of table test is https://github.com/aquasecurity/tracee/blob/main/pkg/server/server_test.go#L22

File: pkg/events/parse/params_test.go (hunk 1 above)
Comment (ShubhamPalriwala) [reply pkg/events/parse/params_test.go:10]: Oh okay! Sorry for not following the format. On it!

**Comment by ShubhamPalriwala** [comment]
Hey @josedonizetti, Took your feedback to the best of my knowledge and have pushed the tests that now cover
- All 3 cases for all the test specs
    - valid (checking for value)
    - invalid (checking for error message)
    - none (checking for error message)

Looking forward to your feedback (would like to apologise beforehand if I've already made some dumb mistakes)

File: pkg/events/parse/params_test.go [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+						Name: tt.arg.Name,
+						Type: tt.arg.Type,
+					},
+					Value: tt.arg.Value,
+				},
+			}}
+			val, err := ArgInt32Val(&e, tt.name)
+			if tt.errorMessage != "" {
+				assert.Equal(t, tt.errorMessage, err.Error())
+				assert.Error(t, err)
```
Comment (josedonizetti) [inline pkg/events/parse/params_test.go:67]: Let's check if there is an error before checking for the message. 

```suggestion
				assert.Error(t, err)
				assert.Equal(t, tt.errorMessage, err.Error())
```

File: pkg/events/parse/params_test.go [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+			} else {
+				assert.Equal(t, tt.expectedValue, val)
+				assert.NoError(t, err)
```
Comment (josedonizetti) [inline pkg/events/parse/params_test.go:70]: check no error before checking expected value
```suggestion
				assert.NoError(t, err)
				assert.Equal(t, tt.expectedValue, val)
```

**Review by josedonizetti — COMMENTED** [review]
@ShubhamPalriwala Thank you for making the changes. Some minor things to address on the tests, and it will be food to go. Thank you for working on it!

File: pkg/events/parse/params_test.go [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			e := trace.Event{Args: []trace.Argument{
+				{
+					ArgMeta: trace.ArgMeta{
+						Name: tt.arg.Name,
+						Type: tt.arg.Type,
+					},
+					Value: tt.arg.Value,
+				},
+			}}
```
[the remark this answers is not in the document]
Comment (ShubhamPalriwala) [reply pkg/events/parse/params_test.go:63]: Done!

File: pkg/events/parse/params_test.go (hunk 2 above)
Comment (ShubhamPalriwala) [reply pkg/events/parse/params_test.go:67]: Done!

File: pkg/events/parse/params_test.go (hunk 3 above)
Comment (ShubhamPalriwala) [reply pkg/events/parse/params_test.go]: Done!

**Review by josedonizetti — APPROVED** [review]
@ShubhamPalriwala thank you! :)

**Pull request merged**